test_api failing

FAILED tests/test_api/test_users_api.py::test_retrieve_user_access_allowed - assert 403 == 200
FAILED tests/test_api/test_users_api.py::test_update_user_email_access_allowed - assert 403 == 200
FAILED tests/test_api/test_users_api.py::test_delete_user - assert 403 == 204
FAILED tests/test_api/test_users_api.py::test_delete_user_does_not_exist - assert 403 == 404
FAILED tests/test_api/test_users_api.py::test_update_user_github - assert 403 == 200
FAILED tests/test_api/test_users_api.py::test_update_user_linkedin - assert 403 == 200
FAILED tests/test_api/test_users_api.py::test_list_users_as_admin - assert 403 == 200
FAILED tests/test_api/test_users_api.py::test_list_users_as_manager - assert 403 == 200

--- a/app/dependencies.py
+++ b/app/dependencies.py
@@ -8,6 +8,7 @@ from app.services.email_service import EmailService
 from app.services.jwt_service import decode_token
 from settings.config import Settings
 from fastapi import Depends
+import logging
 
 def get_settings() -> Settings:
     """Return application settings."""
@@ -37,10 +38,13 @@ def get_current_user(token: str = Depends(oauth2_scheme)):
     )
     payload = decode_token(token)
     if payload is None:
+        print(f"Error retrieving user: {e}")
         raise credentials_exception
     user_id: str = payload.get("sub")
-    user_role: str = payload.get("role")
+    user_role = payload.get("role", "").split('.')[-1] 
+    print(f"User role from token: {user_role}")
     if user_id is None or user_role is None:
+        #print(f"Error retrieving user: {e}")
         raise credentials_exception
     return {"user_id": user_id, "role": user_role}
 

--- a/app/routers/user_routes.py
+++ b/app/routers/user_routes.py
@@ -33,6 +33,9 @@ from app.services.jwt_service import create_access_token
 from app.utils.link_generation import create_user_links, generate_pagination_links
 from app.dependencies import get_settings
 from app.services.email_service import EmailService
+import logging
+
+
 router = APIRouter()
 oauth2_scheme = OAuth2PasswordBearer(tokenUrl="login")
 settings = get_settings()
@@ -50,9 +53,13 @@ async def get_user(user_id: UUID, request: Request, db: AsyncSession = Depends(g
         db: Dependency that provides an AsyncSession for database access.
         token: The OAuth2 access token obtained through OAuth2PasswordBearer dependency.
     """
+    #print(f"Access attempt by user with role: {current_user['role']} to fetch user ID: {user_id}")
+
     user = await UserService.get_by_id(db, user_id)
     if not user:
+        #print(f"User {user_id} not found")
         raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="User not found")
+    
 
     return UserResponse.model_construct(
         id=user.id,

--- a/app/services/jwt_service.py
+++ b/app/services/jwt_service.py
@@ -3,12 +3,13 @@ from builtins import dict, str
 import jwt
 from datetime import datetime, timedelta
 from settings.config import settings
+import logging
 
 def create_access_token(*, data: dict, expires_delta: timedelta = None):
     to_encode = data.copy()
     # Convert role to uppercase before encoding the JWT
     if 'role' in to_encode:
-        to_encode['role'] = str(to_encode['role']).upper()
+        to_encode['role'] = str(to_encode['role']).split('.')[-1].upper()
     expire = datetime.utcnow() + (expires_delta if expires_delta else timedelta(minutes=settings.access_token_expire_minutes))
     to_encode.update({"exp": expire})
     encoded_jwt = jwt.encode(to_encode, settings.jwt_secret_key, algorithm=settings.jwt_algorithm)
@@ -17,6 +18,8 @@ def create_access_token(*, data: dict, expires_delta: timedelta = None):
 def decode_token(token: str):
     try:
         decoded = jwt.decode(token, settings.jwt_secret_key, algorithms=[settings.jwt_algorithm])
+        print(f"Decoded token: {decoded}")
         return decoded
     except jwt.PyJWTError:
+        print(f"Token decoding failed: {e}")
         return None

--- a/tests/test_api/test_users_api.py
+++ b/tests/test_api/test_users_api.py
@@ -33,6 +33,7 @@ async def test_retrieve_user_access_denied(async_client, verified_user, user_tok
 async def test_retrieve_user_access_allowed(async_client, admin_user, admin_token):
     headers = {"Authorization": f"Bearer {admin_token}"}
     response = await async_client.get(f"/users/{admin_user.id}", headers=headers)
+    print("response.status_code : " , response.status_code)
     assert response.status_code == 200
     assert response.json()["id"] == str(admin_user.id)
 

